EventSubscriber/NavigationInvalidationSubscriber.php

```diff
@@ -116,12 +116,14 @@ public function collectNavigationContextBeforeRemovingLocale(RemoveLocaleEvent $
         $this->collectNavigationContexts($path, $event->getLocale());
     }
 
-    public function collectNavigationContexts(string $path, string $locale): void
+    public function collectNavigationContexts(string $path, ?string $locale): void
     {
         $defaultNode = $this->defaultSession->getNode($path);
         $liveNode = $this->liveSession->getNode($path);
 
-        $propertyName = $this->propertyEncoder->localizedContentName('navContexts', $locale);
+        $propertyName = $locale ?
+            $this->propertyEncoder->localizedContentName('navContexts', $locale) :
+            $this->propertyEncoder->contentName('navContexts');
```

Comment:
are we doing something similar in sulu core?

Reply:
as discusswe if no locale is given all locales nees to be purged. E.g. moving of a page.